### PR TITLE
FindGridSpots: add unitest that checks the correct rotation is returned for rotated spots

### PR DESCRIPTION
**Pull request by tepals**
…d for rotated spots.

### CODE REVIEW

File: src/odemis/acq/align/test/spot_test.py [hunk 1]
```diff
@@ -290,6 +292,32 @@ def test_find_grid_with_shear(self):
...
+        rotated_indices += 10  # make sure each spot is at least 10 pixels away from the border
+        max_idx = int(numpy.max(rotated_indices))
+        rotated_image = numpy.zeros((max_idx + 10, max_idx + 10))
+        # set a grid of 8 by 8 rotated points to 1
+        for indice in rotated_indices:
+            row = int(round(indice[0]))
+            column = int(round(indice[1]))
+            rotated_image[row][column] = 1
+        spot_coordinates, translation, scaling, rotation, shear = spot.FindGridSpots(rotated_image, (8, 8))
+        self.assertAlmostEqual(rotation, -img_rotation, places=2)
```
Comment (nandishjpatel) [inline src/odemis/acq/align/test/spot_test.py:319]: Maybe add a comment for the `-img_rotation` sign?

File: src/odemis/acq/align/test/spot_test.py [hunk 2]
```diff
@@ -290,6 +292,32 @@ def test_find_grid_with_shear(self):
...
+        indices = numpy.where(image)
+        indices = numpy.column_stack((indices[0], indices[1]))
+        img_rotation = math.radians(10)
+        transform = ScalingTransform(rotation=img_rotation)
+        # get the rotated indices
+        rotated_indices = transform.apply(indices)
+        rotated_indices -= numpy.min(rotated_indices)
+        rotated_indices += 10  # make sure each spot is at least 10 pixels away from the border
+        max_idx = int(numpy.max(rotated_indices))
+        rotated_image = numpy.zeros((max_idx + 10, max_idx + 10))
```
Comment (nandishjpatel) [inline src/odemis/acq/align/test/spot_test.py:312]:
```suggestion
        max_idx = int(numpy.max(rotated_indices))
        max_idx += 10
        rotated_image = numpy.zeros((max_idx, max_idx))
```

Comment (pieleric) [reply src/odemis/acq/align/test/spot_test.py:312]: Then, just write `max_idx = int(numpy.max(rotated_indices)) + 10`
But maybe max_idx is not a good name anymore?
I think the original code was pretty fine.

**Review by pieleric — APPROVED** [review]

**Review by nandishjpatel — APPROVED** [review]

File: src/odemis/acq/align/test/spot_test.py (hunk 2 above)
Comment (tepals) [reply src/odemis/acq/align/test/spot_test.py:312]: I'll leave it like it was

**Pull request merged**